core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala

```diff
@@ -1544,6 +1544,16 @@ trait DeltaErrorsBase
     )
   }
 
+  def configureSparkSessionWithExtensionAndCatalog(
```

Comment:
Could you remove the above `def configureSparkSessionWithExtensionAndCatalog(originalException: Throwable): Throwable` and fix existing usages? We don't need to keep two versions of such method.

Reply:
Sure. I found two usages of this method. Will remove this method and refactor the existing function calls to use `configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable]) instead.

Reply:
Actually, `configureSparkSessionWithExtensionAndCatalog(originalException: Throwable)` throws `DeltaAnalysisException` and `configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable])` throws `DeltaIllegalStateException`

Are we okay throwing `DeltaIllegalStateException` from `AnalysisHelper.isExtensionOrCatalogError()`?

Reply:
Let's use `DeltaAnalysisException` instead. Mostly of time when we create DeltaLog for the first time, it will be in the analysis phase.

Reply:
I tried using `configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable])` that throws `DeltaAnalysisException` and if the check if the spark config not found, it actually fails with `java.util.concurrent.ExecutionException` which is thrown by the `getDeltaLogFromCache()` called from the `DeltaLog` constructor. 

Because of this my unit test fails with:
```
org.scalatest.exceptions.TestFailedException: Expected exception org.apache.spark.sql.delta.DeltaAnalysisException to be thrown, but java.util.concurrent.ExecutionException was thrown.

Caused by: java.util.concurrent.ExecutionException: org.apache.spark.sql.delta.DeltaAnalysisException: This Delta operation requires the SparkSession to be configured with the
DeltaSparkSessionExtension and the DeltaCatalog. Please set the necessary
configurations when creating the SparkSession as shown below.

  SparkSession.builder()
    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
    ...
    .getOrCreate()

```
Any idea how to make `intercept[DeltaAnalysisException] ` work in this case?